src/client.ts

```diff
@@ -63,9 +64,12 @@ export class Connection {
   private heartbeat: Heartbeat
   private consumerId = 0
   private consumers = new Map<number, StreamConsumer>()
+  private producers = new Map<number, StreamProducer>()
+  private producerConnections = new Map<number, Client>()
```

Comment:
Instead of saving the `connection/client` instance on a separate map, we could add a getter method on the producer (the info is already there)

Reply:
Yeah, but the producer is something we hand to the user and I didn't want to let them take a hold of the underlying client D: